src/node-sdk/src/index.ts

```diff
@@ -1676,6 +1679,101 @@ export function deleteUserGrantAs(userId: string, applicationId: string, apiId:
     return apiDelete(`grants/${userId}/applications/${applicationId}/apis/${apiId}`, asUserId, callback);
 }
 
+// ======= ACCESS TOKEN MANAGEMENT =======
+
+/*
+ * Please note that this is not intended for public use; it's used by the Authorization
+ * server to keep track of Kong's access and refresh tokens, and to be able to support
+ * retrieving information on them, for e.g. subsequent deletion or similar. This may
+ * change in the future, but for now, it's how it works.
+ */
+
+/**
+ * Registers access token data with wicked's API for future reference.
+ * 
+ * @param tokenData Token to register with wicked's API
+ */
+export function registerAccessToken(tokenData: WickedAccessToken): Promise<any>;
+export function registerAccessToken(tokenData: WickedAccessToken, callback: ErrorCallback);
+export function registerAccessToken(tokenData: WickedAccessToken, callback?: ErrorCallback) {
+    return apiPost('accesstokens', tokenData, callback, null);
+}
+
+/**
+ * Retrieves information on a given access token. Is expected to return a collection with one single item,
+ * or an empty collection, if the access token is not found.
+ * 
+ * @param accessToken Access token to retrieve information on
+ */
+export function getAccessToken(accessToken: string): Promise<WickedCollection<WickedAccessToken>>;
+export function getAccessToken(accessToken: string, callback: Callback<WickedCollection<WickedAccessToken>>);
+export function getAccessToken(accessToken: string, callback?: Callback<WickedCollection<WickedAccessToken>>) {
+    return apiGet(`accesstokens?access_token=${encodeURIComponent(accessToken)}`, callback, null);
+}
+
+/**
+ * Retrieves information on the access token associated with the given refresh token. Is expected
+ * to return a collection with one single item (or none, if not found).
+ * 
+ * @param refreshToken Refresh token to retrieve information on
+ */
+export function getAccessTokenByRefreshToken(refreshToken: string): Promise<WickedCollection<WickedAccessToken>>;
+export function getAccessTokenByRefreshToken(refreshToken: string, callback: Callback<WickedCollection<WickedAccessToken>>);
+export function getAccessTokenByRefreshToken(refreshToken: string, callback?: Callback<WickedCollection<WickedAccessToken>>) {
+    return apiGet(`accesstokens?refresh_token=${encodeURIComponent(refreshToken)}`, callback, null);
+}
+
+/**
+ * Retrieves all access tokens associated with the given authenticated_userid string. This can result in
+ * multiple items being returned, or none, if the user does not have any associated access tokens currently.
+ * 
+ * @param authenticatedUserId authenticated_userid string to retrieve access tokens for
+ */
+export function getAccessTokensByAuthenticatedUserId(authenticatedUserId: string): Promise<WickedCollection<WickedAccessToken>>;
+export function getAccessTokensByAuthenticatedUserId(authenticatedUserId: string, callback: Callback<WickedCollection<WickedAccessToken>>);
```

Comment:
For one of our use case we inject product license in x-authenticated-userid header (in addition to userid), so I am assuming this call would work for that case as well

Reply:
Yes-ish. The idea is that you can use the value of the header you get to the backend to query wicked for all current access tokens this exact user has. But if you have altering headers for the users, this will obviously not work. The product license sounds like it _perhaps_ rather should be in the `x-authenticated-scope` header.